### PR TITLE
Make ImageView clickable/linkable in Python Panels

### PR DESCRIPTION
**Pull request by Br2850**
## What changes are proposed in this pull request?

Give images the ability to:

- Open links (via href)
- Call operators w/ params
- Added new `.img()` built in method to allow Python Panels a shortcut to define images

Updated docstrings and examples to correctly identify how to use `.img()` and `ImageView`

## How is this patch tested? If it is not, please explain why.

See screen recording below w/ code

```
 def on_load(self, ctx):
        ctx.panel.state.image_link = "https://i.imgur.com/TGEZj1Rl.jpg"

schema = {
            "height": "200px",
            "width": "200px",
            "alt": "My image alt text",
            "href": "https://voxel51.com",
            "operator": self.do_something,
            "prompt": True,
            "params": {"foo": "bar"},
        }
        panel.define_property(
            "image_link",
            types.String(),
            view=types.ImageView(**schema),
        )
```

https://github.com/user-attachments/assets/d68c1a5c-00af-4c51-9dfc-17c9683b3ee6

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced `ImageView` component with additional properties for improved interactivity.
	- Introduced click handling that supports complex user interactions and external URL navigation.
	- Added a new `img` method to define image properties for display.

- **Documentation**
	- Updated documentation for the `ImageView` class to include new parameters, improving clarity for developers.

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx [hunk 1]
```diff
@@ -16,6 +45,7 @@
         height={height}
         width={width}
         alt={alt}
+        onClick={onClick}
```
[the remark this answers is not in the document]
Comment (Br2850) [reply app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx:48]: Added cursor on hover if href for accessibility

File: app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx [hunk 2]
```diff
@@ -2,12 +2,41 @@ import { Box } from "@mui/material";
...
+      handleClick(panelId, {
+        params: params,
+        operator,
+        prompt,
+      });
+    }
+
+    if (href) {
+      window.open(href, "_blank");
+    }
```
[the remark this answers is not in the document]
Comment (Br2850) [reply app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx:37]: https://stackoverflow.com/questions/50709625/link-with-target-blank-and-rel-noopener-noreferrer-still-vulnerable

File: fiftyone/operators/types.py [hunk 3]
```diff
@@ -1777,6 +1777,15 @@ def execute():
             return {"image": "https://voxel51.com/your/image.png"}
 
         def resolve_output(self, ctx):
+            schema = {
+                "height": "100px",
+                "width": "100px",
+                "alt": "My image alt text",
+                "href": "https://voxel51.com",
+                "operator": self.do_something,
```
Comment (ritch) [inline fiftyone/operators/types.py:1785]: This will only work in a python panel, `resolve_output` is only for `foo.Operator`s.

Comment (ritch) [reply fiftyone/operators/types.py:1785]: Instead you could say:

```python
"operator": "@my/plugin/my_operator"
```

File: app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx [hunk 4]
```diff
@@ -16,6 +45,8 @@ export default function ImageView(props) {
         height={height}
         width={width}
         alt={alt}
+        onClick={onClick}
+        style={{ cursor: href ? "pointer" : "default" }} // Change cursor based on href
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx:49]: We might want to make this a setting as well. Its weird to not get the `pointer` when the intention is that the image is clickable. Maybe something like:

```python
panel.img('img', ..., pointer_on_hover=True)
```

File: app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx [hunk 5]
```diff
@@ -2,12 +2,41 @@ import { Box } from "@mui/material";
...
   const imageURI = data ?? schema?.default;
 
+  const panelId = usePanelId();
+  const handleClick = usePanelEvent();
+
+  const onClick = () => {
+    // ordering matters: execute operator first then redirect if defined
+
+    if (operator) {
+      handleClick(panelId, {
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx:28]: if an operator is provided we should call `handleClick` and provide a callback.

```typescript
    triggerEvent(panelId, {
      operator,
      params,
      prompt: props.prompt, // we could also support this
      callback: (result: OperatorResult) => {
        if (href) window.open(href, "_blank");
      },
    });
```

File: fiftyone/operators/types.py [hunk 6]
```diff
@@ -1785,6 +1794,15 @@ def resolve_output(self, ctx):
                 view=types.ImageView(),
             )
             return types.Property(outputs)
+
+    Args:
+        height (None): the height of the image
+        width (None): the width of the image
+        alt (None): the alt text of the image
+        href (None): the href of the image
+        operator (None): the name of the callable operator to execute when the image is clicked
```
Comment (ritch) [inline fiftyone/operators/types.py:1803]: Please add a `.img()` util with `on_click` similar to the `.btn()` util.

File: fiftyone/operators/types.py (hunk 3 above)
Comment (Br2850) [reply fiftyone/operators/types.py:1785]: Updated example to make sure Python Panel reference & Operator reference is noted 👍🏽

File: app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx (hunk 5 above)
Comment (Br2850) [reply app/packages/core/src/plugins/SchemaIO/components/ImageView.tsx:28]: Moved additionally functionality to callback to strictly make sure execution is done in order

**Review by ritch — APPROVED** [review]

**Pull request merged**